evalml/pipelines/components/transformers/preprocessing/transform_primitive_components.py

```diff
@@ -0,0 +1,180 @@
+from abc import abstractmethod
+
+import featuretools as ft
+
+from evalml.pipelines.components.transformers.transformer import Transformer
+from evalml.utils import infer_feature_types
+
+
+class _ExtractFeaturesWithTransformPrimitives(Transformer):
```

Comment:
Added a parent class to minimize the code duplication that comes from the following pattern:

1. specify some ft primitives at class definition
2. Run dfs to extract features with those primitives
3. Delete original columns

Maybe this could be used for the TextFeaturizer later as well.

Reply:
I think this is a great idea

Reply:
❤️  this!